sources/editor/Stride.Assets.Presentation/AssetEditors/Gizmos/ViewportGridGizmo.cs

```diff
@@ -167,14 +176,14 @@ private Material CreateColoredTextureMaterial(Texture texture, float textureScal
             });
         }
 
-        protected override void UpdateBase(Color3 gridColor, float alpha, int gridAxisIndex, float sceneUnit)
+        private int UpdateGrid(Color3 gridColor, float alpha, int gridAxisIndex, float sceneUnit, int gridIndex)
```

Comment:
Why are you not reusing `gridAxisIndex` for `gridIndex`?

Reply:
I'll try to simplify "AxisSelection" first and then refactor this part later.